src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/ReferingToObjectAndReassigningItInTheSameStatement.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    /// <summary>
+    /// ReferenceChanged: Prevent objects from being referenced in statements where they are reassigned
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+
+    public sealed class ReferingToObjectAndReassigningItInTheSameStatement : DiagnosticAnalyzer
```

Comment:
Another possibility:

```suggestion
    public sealed class ReferencingAndReassigningObjectInSameStatement : DiagnosticAnalyzer
```

Reply:
Great name, thanks!